dcicutils/s3_utils.py

```diff
@@ -24,16 +25,70 @@ class s3Utils(object):  # NOQA - This class name violates style rules, but a lot
     RAW_BUCKET_TEMPLATE = "elasticbeanstalk-%s-files"
     BLOB_BUCKET_TEMPLATE = "elasticbeanstalk-%s-blobs"
 
-    def __init__(self, outfile_bucket=None, sys_bucket=None, raw_file_bucket=None,
-                 blob_bucket=None, env=None):
+    @staticmethod
+    def verify_and_get_env_config(s3_client, global_bucket: str):
+        """ Verifies the S3 environment from which the env config is coming from, and returns the S3-based env config
+            Throws exceptions if the S3 bucket is unreachable, or an env based on the name of the global S3 bucket
+            is not present.
         """
-        if we pass in env set the outfile and sys bucket from the environment
+        head_response = s3_client.head_bucket(Bucket=global_bucket)
+        status = head_response['ResponseMetadata']['HTTPStatusCode']  # should be 200; raise error for 404 or 403
+        if status != 200:
+            raise Exception('Could not access GLOBAL_BUCKET_ENV {global_bucket}: status: {status}'.format(
+                global_bucket=global_bucket, status=status))
+        # list contents of global bucket, look for a match with the global bucket name
+        list_response = s3_client.list_objects_v2(Bucket=global_bucket)
+        # no match, raise exception
+        if list_response['KeyCount'] < 1:
+            raise Exception('No config objects found in global bucket {global_bucket}'.format(
```

Comment:
Same issue here again, although this suggests the code I wrote for GlobalBucketAccessFailure above is not general enough, so I could rewrite it:

```
class GlobalBucketException(S3UtilsException):
    MESSAGE = "Problem with"
    def __init__(self, **kwargs):
        for k, v in kwargs.items():
            setattr(self, k, v)
        super().__init__(self.MESSAGE.format(**kwargs))

class GlobalBucketAccessFailure(GlobalBucketException):
    MESSAGE = "Could not access GLOBAL_BUCKET_ENV {bucket}: status: {status}"
    def __init__(self, bucket, status):
        super().__init__(bucket=bucket, status=status)
```
so that we can add
```
class GlobalBucketMissingConfig(GlobalBucketException):
    MESSAGE = "No config objects found in global bucket {bucket}."
    def __init__(self, bucket):
        super().__init__(bucket=bucket)
```
and we can write (untested):
```
    raise GlobalBucketMissingConfig(bucket=global_bucket)
```